tmt/frameworks/__init__.py

```diff
@@ -108,15 +108,31 @@ def get_pull_options(
 
         return []
 
+    @classmethod
+    def results_to_subresults(cls, results: list[tmt.result.Result]) -> list[tmt.result.SubResult]:
+        """ Convert given results to tmt subresults """
+
+        return [
```

Comment:
TODO: Try to find a better way to convert `tmt.Result` to `tmt.result.SubResult`, e.g. using their `to_spec/from_spec`. CC: @happz

Reply:
I'd probably move this method into `Result` class, on the assumption that `Result` knows best how it should be "lowered" into `Subresult`. Its presence here looks like an implementation leak, suddenly frameworks own the secret sauce, instead of `tmt.result`.

Reply:
Great tip. Makes sense - moved to `tmt.Result.to_subresult`.